code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -685,3 +747,169 @@
+
+// ***************************************
+// *********** Displacement
+// ***************************************
+
+/datum/action/ability/xeno_action/displacement
+	name = "Displacement"
+	action_icon_state = "hunter_invisibility"
+	action_icon = 'icons/Xeno/actions/hunter.dmi'
+	desc = "Physically disappear, become incorporeal until you decide to reappear somewhere else, reappearing on lighted areas will disorient you and flicker the lights."
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOMORPH_HUNTER_DISPLACEMENT,
+	)
+	use_state_flags = ABILITY_USE_SOLIDOBJECT
+
+/datum/action/ability/xeno_action/displacement/action_activate()
+	var/mob/living/carbon/xenomorph/xenomorph_owner = owner
+	xenomorph_owner.change_form()
+
+/mob/living/carbon/xenomorph/hunter/assassin/change_form()
+	if(!loc_weeds_type)
+		balloon_alert(src, "We need to be on weeds.")
+		return
+	wound_overlay.icon_state = "none"
+	var/turf/whereweat = get_turf(src)
+	if(status_flags & INCORPOREAL) //will alert if the xeno will get disoriented due lit turf, if phasing in.
+		if(whereweat.get_lumcount() > 0.3) //is it a lit turf
+			balloon_alert(src, "We will be disoriented and sensed in this light.") //so its more visible to xeno.
+			//Marines can sense the manifestation if it's in lit-enough turf nearby.
+			visible_message(span_highdanger("Something begins to manifest nearby!"), span_xenohighdanger("We begin to manifest in the light... talls sense us!"))
+	else
+		if(whereweat.get_lumcount() > 0.4) //cant shift out a lit turf.
+			balloon_alert(src, "We need a darker spot.") //so its more visible to xeno.
+			return
+	if(do_after(src, 3 SECONDS, IGNORE_HELD_ITEM, src, BUSY_ICON_BAR, NONE, PROGRESS_GENERIC)) //dont move
+		do_change_form()
+
+///Finish the form changing of the hunter and give the needed stats
+/mob/living/carbon/xenomorph/hunter/assassin/proc/do_change_form()
+	playsound(get_turf(src), 'sound/effects/alien/new_larva.ogg', 25, 0, 1)
+	if(status_flags & INCORPOREAL)
+		var/turf/whereweat = get_turf(src)
+		if(whereweat.get_lumcount() > 0.3) //is it a lit turf
+			balloon_alert(src, "Light disorients us!")
+			adjust_stagger(6 SECONDS)
+			add_slowdown(4)
+		for(var/obj/machinery/light/lightie in range(rand(7,10), whereweat))
+			lightie.set_flicker(2 SECONDS, 1.5, 2.5, rand(1,2))
+		status_flags = initial(status_flags)
+		resistance_flags = initial(resistance_flags)
+		pass_flags = initial(pass_flags)
+		density = TRUE
+		REMOVE_TRAIT(src, TRAIT_HANDS_BLOCKED, src)
+		alpha = 255
+		update_wounds()
+		update_icon()
+		update_action_buttons()
+		return
+	var/turf/whereweat = get_turf(src)
+	for(var/obj/machinery/light/lightie in range(rand(7,10), whereweat))
+		lightie.set_flicker(2 SECONDS, 1, 2, rand(1,2))
+	ADD_TRAIT(src, TRAIT_HANDS_BLOCKED, src)
+	status_flags = INCORPOREAL
+	alpha = 0
+	resistance_flags = BANISH_IMMUNE
+	pass_flags = PASS_MOB|PASS_XENO
+	density = FALSE
+	update_wounds()
+	update_icon()
+	update_action_buttons()
```

Comment:
why is this an assassin mob proc instead of the ability?

Reply:
that is how it is done at hivemind and i just did it like that, guess i can change it.